docs: update docs to use MyST

pfackeldey opened the pull request:
This PR updates the docs to:
- use MyST markdown instead of restructured text (except for API ref; they're a bit fragile right now so I kept them as they are)
- use a modern theme (pydata)
- reduce ~200 bugs/typos from existing doc strings
- add a `User Guide` section that can be filled with more 'hands-on' docs

pfackeldey committed:
start of updating docs to use MyST

pfackeldey committed:
start of updating docs to use MyST

lgray commented:
I'll poke at this later today or tomorrow to see if I can get RTD consuming it correctly.

lgray committed:
fix some minor complaints during docs building

lgray committed:
update filename

lgray committed:
.md not .rst now :-)

lgray committed:
README still .rst

ikrommyd committed:
Merge branch 'master' into pfackeldey/overhaul_docs

ikrommyd committed:
Merge branch 'master' into pfackeldey/overhaul_docs

pfackeldey committed:
Merge branch 'master' into pfackeldey/overhaul_docs

pfackeldey committed:
add first bulk of docs content

pfackeldey committed:
resolve

pfackeldey committed:
resolve

pfackeldey commented:
Hi @ikrommyd and @lgray,
I've added quite some content to the docs (with the help of Codex :P). Afterwards I went through manually and adjusted/fixed obvious hallucinations or things that are non-sense, and added some stuff by hand. I do not intend to update the example notebooks with this PR. 
Please have a look how you like the docs and the content. Let me know if you want something in addition or different. In principle I would consider this PR done (for now).

pfackeldey committed:
remove non-sense section

ikrommyd committed:
Merge branch 'master' into pfackeldey/overhaul_docs

ikrommyd committed:
complete notebook overhaul

ikrommyd committed:
remove workqueue from docs

ikrommyd committed:
forgot this

ikrommyd committed:
update installation

ikrommyd committed:
user guide updates

ikrommyd committed:
fix doc building warnings

ikrommyd committed:
various things

ikrommyd committed:
use or in docstrings

ikrommyd committed:
couple more fixes

ikrommyd reviewed (approved):
I can't manually review this large diff of course line by line but this PR does not touch actual code. It only touches docstrings and docs. No tests touches too. I therefore think that we can trust it cannot break user code.

I myself threw some LLMs at overall doc formatting and docstring formatting of the codebase too. The coffea docs now look much better than they used to so I think it's safe to merge this.

ikrommyd commented:
Thanks a lot @pfackeldey!

It's already a pretty huge PR. The docs are not perfect but this is already a great improvement. I'm merging it and we can go from there in separate PRs for improvements.
